chia/wallet/puzzles/did_innerpuz.clvm

```diff
@@ -132,28 +113,16 @@
 
   ;Spend modes:
   ;0 = recovery
-  ;1 = create messages and recreate singleton
-  ;2 = transfer current DID to a new wallet
+  ;1 = run the P2_PUZZLE
 
   ;MAIN
   (if mode
-    ; mode 1 - create messages and recreate singleton
-    (if (= mode 1)
-      ; Send normal messages
-      (c (list CREATE_COIN new_inner_puzhash new_amount (list new_inner_puzhash)) (c (list AGG_SIG_ME (get_pubkey P2_PUZZLE) (sha256tree1 (list new_inner_puzhash new_amount))) (a P2_PUZZLE p2_solution)))
-      ; Transfer DID
-      (if (= (sha256tree1 recovery_list_reveal) RECOVERY_DID_LIST_HASH)
-         (c (list CREATE_COIN (calculate_did_puzzle_hash new_inner_puzhash RECOVERY_DID_LIST_HASH NUM_VERIFICATIONS_REQUIRED SINGLETON_STRUCT DID_MOD_HASH METADATA) new_amount (list new_inner_puzhash)) (c (list AGG_SIG_ME (get_pubkey P2_PUZZLE) (sha256tree1 (list new_inner_puzhash new_amount recovery_list_reveal))) (a P2_PUZZLE p2_solution)))
-         (x)
-      )
-    )
+    ; mode 1 - run P2_PUZZLE
+    (a P2_PUZZLE new_amount_or_p2_solution)
```

Comment:
Same as above, I think we need to reveal the recovery list in the transfer case. Better to add a hash check at higher layer, in order to saving cost and prevent the sender revealing a wrong recovery list.